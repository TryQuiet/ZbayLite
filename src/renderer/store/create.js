import { applyMiddleware, createStore, compose } from 'redux'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'
import createDebounce from 'redux-debounced'

import reducers from './reducers'
import { errorsMiddleware } from './middlewares'

const composer = (enhancers) => {
  if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      process.env.NODE_ENV === 'development') {
    return window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__(enhancers)
  }
  return compose(enhancers)
}

export default ({
  initialState
} = {}) => createStore(
  reducers,
  initialState,
  composer(applyMiddleware(...[errorsMiddleware, createDebounce(), thunk, promise()]))
)
