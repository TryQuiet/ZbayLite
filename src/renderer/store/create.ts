import { applyMiddleware, createStore } from 'redux'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'
import createDebounce from 'redux-debounced'
import createSagaMiddleware from 'redux-saga'
import rootSaga from '../sagas/index.saga'
import { composeWithDevTools } from 'redux-devtools-extension'

import reducers from './reducers'
import { errorsMiddleware } from './middlewares'

const sagaMiddleware = createSagaMiddleware()

export default (initialState = {}) => {
  const store = createStore(
    reducers,
    initialState,
    composeWithDevTools(
      applyMiddleware(...[errorsMiddleware, createDebounce(), sagaMiddleware, thunk, promise()])
    )
  )
  sagaMiddleware.run(rootSaga)
  return store
}
