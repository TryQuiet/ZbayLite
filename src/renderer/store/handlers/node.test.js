/* eslint import/first: 0 */
import create from '../create'
import nodeHandlers from './node'
import nodeSelectors from '../selectors/node'

describe('Node reducer handles', () => {
  let store = null
  beforeEach(() => {
    store = create()
    jest.clearAllMocks()
  })

  const assertStoreState = () => expect(
    nodeSelectors.node(store.getState())
  ).toMatchSnapshot()

  describe('epics -', () => {
    describe('getStatus', () => {
      it('when successfull', async () => {
        await store.dispatch(nodeHandlers.epics.getStatus())
        assertStoreState()
      })

      it('when node is down', async () => {
        try {
          await store.dispatch(nodeHandlers.epics.getStatus())
        } catch (err) {}
        assertStoreState()
      })
    })
  })
})
