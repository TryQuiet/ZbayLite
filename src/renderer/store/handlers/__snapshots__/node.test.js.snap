// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node reducer handles epics - getStatus when node is down 1`] = `
Object {
  "bootstrappingMessage": "",
  "connections": "0",
  "currentBlock": "0",
  "errors": [TypeError: _electron.ipcRenderer.send is not a function],
  "fetchingStatus": Object {
    "currentSlide": 0,
    "fetchingEndTime": Object {
      "hours": null,
      "minutes": null,
      "seconds": null,
    },
    "fetchingSpeed": null,
    "fetchingStatus": "",
    "guideStatus": false,
    "isFetching": false,
    "isRescanningInitialized": false,
    "isRescanningMonitorStarted": false,
    "part": "",
    "rescanningProgress": 0,
    "sizeLeft": 0,
  },
  "isRescanning": false,
  "isTestnet": null,
  "latestBlock": "999999",
  "loading": false,
  "startedAt": null,
  "status": "down",
}
`;

exports[`Node reducer handles epics - getStatus when successfull 1`] = `
Object {
  "bootstrappingMessage": "",
  "connections": "0",
  "currentBlock": "0",
  "errors": [TypeError: _electron.ipcRenderer.send is not a function],
  "fetchingStatus": Object {
    "currentSlide": 0,
    "fetchingEndTime": Object {
      "hours": null,
      "minutes": null,
      "seconds": null,
    },
    "fetchingSpeed": null,
    "fetchingStatus": "",
    "guideStatus": false,
    "isFetching": false,
    "isRescanningInitialized": false,
    "isRescanningMonitorStarted": false,
    "part": "",
    "rescanningProgress": 0,
    "sizeLeft": 0,
  },
  "isRescanning": false,
  "isTestnet": null,
  "latestBlock": "999999",
  "loading": false,
  "startedAt": null,
  "status": "down",
}
`;
